docs/static/security/api-keys.asciidoc

```diff
@@ -2,8 +2,145 @@
 [[ls-api-keys]]
 ==== Grant access using API keys
 
-
 Instead of using usernames and passwords, you can use API keys to grant
 access to {es} resources. You can set API keys to expire at a certain time,
 and you can explicitly invalidate them. Any user with the `manage_api_key`
 or `manage_own_api_key` cluster privilege can create API keys.
+
+Note that API keys are tied to the cluster they are created in. If you are
+sending output to different clusters, be sure to create the correct kind of API
+key. 
+
+NOTE: For security reasons, we recommend using a unique API key per {ls} instance.
+You can create as many API keys per user as necessary.
+
+
+[float]
+[[ls-create-api-key]]
+===== Create an API key 
+
+You can create API keys using either the
+{ref}/security-api-create-api-key.html[Create API key API] or the
+{kibana-ref}/api-keys.html[Kibana UI]. This section walks you through creating
+an API key using the {ref}/security-api-create-api-key.html[Create API key API].
+The privileges needed are the same for either approach.
+
+Here is an example that shows how to create an API key for publishing to {es}.
+
+
+[source,console,subs="attributes,callouts"]
+------------------------------------------------------------
+POST /_security/api_key
+{
+  "name": "logstash_host001", <1>
+  "role_descriptors": {
+    "logstash_writer": { <2>
+      "cluster": ["monitor", "read_ilm"],
```

Comment:
was this checked? I believe that if the ilm policy isn't there, which  isn't in a clean installation of elasticsearch, then we'll create an alias  and  push an ilm policy with a template. So we may need `manage_ilm` here too?

Reply:
I have not tested yet. Still scrambling to get structure and content in place.

Reply:
not sure about the roles here, suggestion to move forward and/or defer that in the future to avoid missing train?

Reply:
@jsvd  any suggestions about that to move forward? I am not sure personally about the correct permissions/roles here.

Reply:
@colinsurprenant we can draw from existing docs on setting up security with user/pwd such as https://www.elastic.co/guide/en/logstash/current/ls-security.html#ls-http-auth-basic, where we already show the needed roles for the elasticsearch output, same thing for monitoring (https://www.elastic.co/guide/en/elasticsearch/reference/7.7/built-in-users.html)

there's a component of new territory here in finding out the roles for these use cases, [out of all those built into elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-get-builtin-privileges.html).
A suggestion to find this out it so leverage an elastic cloud account, where security is already set up, and doesn't require creating certs.

Reply:
should we postpone that work or hold this PR up until we get all the roles/permissions right?